### PR TITLE
Refresh signal on Windows

### PR DESCRIPTION
**Pull request by Slava**
Not only enables the client refresh to happen on Windows, also changes the posix signal from SIGUSR2 to SIGHUP.

@glasser, can you please review it?

### CODE REVIEW

File: tools/run-app.js [hunk 1]
```diff
@@ -243,7 +243,8 @@ _.extend(AppProcess.prototype, {
     // Call node
     var child_process = require('child_process');
     var child = child_process.spawn(nodePath, opts, {
-      env: self._computeEnvironment()
+      env: self._computeEnvironment(),
+      stdio: ['pipe', 'pipe', 'pipe', 'ipc']
```
Comment (glasser) [inline tools/run-app.js]: Probably only do this if watchForChanges is set.  ie, if `--once` is passed then we shouldn't create this extra fd.  (which might actually have an effect on whether or not the process terminates if the fd isn't disconnected?)

File: tools/run-app.js [hunk 2]
```diff
@@ -751,7 +744,7 @@ _.extend(AppRunner.prototype, {
 
         // Notify the server that new client assets have been added to the
         // build.
-        appProcess.proc.kill('SIGUSR2');
+        appProcess.proc.send({ refresh: 'client' });
```
Comment (glasser) [inline tools/run-app.js]: The docs say "Emits an 'error' event if the message cannot be sent, for example because the child process has already exited".  (presumably on the proc object?)  now, we do have an error handler on the proc object but it logs "Couldn't spawn process" which is wrong in this case.  see the docs: http://nodejs.org/api/child_process.html#child_process_event_error

File: tools/watch.js [hunk 3]
```diff
@@ -267,7 +267,6 @@ var readDirectory = function (options) {
       }
       throw e;
     }
-    // XXX if we're on windows, I guess it's possible for files to end with '/'.
```
Comment (glasser) [inline tools/watch.js]: this seems unrelated?

**Comment by stubailo** [comment]
Looks good to me other than Glasser's comments.

File: tools/watch.js (hunk 3 above)
Comment (Slava) [inline tools/watch.js]: I deleted an old comment that was concerned about the future windows development. I think the comment is not relevant any more, so I deleted it.

**Comment by Slava** [comment]
thanks, merged

File: tools/run-app.js [hunk 4]
```diff
@@ -242,8 +247,13 @@ _.extend(AppProcess.prototype, {
 
     // Call node
     var child_process = require('child_process');
+    // setup the 'ipc' pipe if further communication between app and proxy is
+    // expected
+    var ioOptions = self.watchForChanges
```
Comment (glasser) [inline tools/run-app.js]: I don't think that `AppProcess` has watchForChanges yet
